Product Management Service/Code/product-service/src/main/java/com/shopkeeper/productservice/controller/ImageController.java

```diff
@@ -1,8 +1,47 @@
 package com.shopkeeper.productservice.controller;
 
+import java.util.List;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.DeleteMapping;
+import org.springframework.web.bind.annotation.GetMapping;
+import org.springframework.web.bind.annotation.PathVariable;
+import org.springframework.web.bind.annotation.PostMapping;
 import org.springframework.web.bind.annotation.RequestMapping;
+import org.springframework.web.bind.annotation.RequestParam;
 import org.springframework.web.bind.annotation.RestController;
+import org.springframework.web.multipart.MultipartFile;
+
+import com.shopkeeper.productservice.dto.ProductImages;
+import com.shopkeeper.productservice.service.ImageService;
 
 @RestController
 @RequestMapping(value = "/images")
-public class ImageController {}
+public class ImageController {
+
+  @Autowired ImageService imageService;
+
+  @GetMapping("/{id}")
+  public ResponseEntity<List<ProductImages>> getImagesByProductId(@PathVariable Integer id) {
```

Comment:
Generally the wrapper type is used if we want our endpoint to allow null values. but in our case, we don't want to pass in null values for IDs as that would be wrong. Thus, we should be going with int instead of Integer.